### PR TITLE
Add Registration Logic to CredentialsManager and Registration layout

### PR DESCRIPTION
**Pull request by ByalykT01**
This PR adds a register method to the CredentialsManager that:

- Registers a new account if the email is not already in use (case-insensitive).
- Returns an error if the email is already registered.
- Uses a Map for storing email-password pairs, normalizing emails to lowercase for comparison.

Testing

- Verified that duplicate emails (e.g., test@te.st and TEST@te.st) are rejected.
- Confirmed successful registration for unique emails.

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/CredentialsManager.kt [hunk 1]
```diff
@@ -3,11 +3,29 @@ package com.example.myapplication
...
 
 class CredentialsManager {
-    val credentials = mutableMapOf<String, String>(
-        //
-    )
+    private val credentials = mutableMapOf<String, String>()
 
-    fun isEmailValid(mail: String): Boolean {
+
+    fun register(email: String, password: String): Pair<Boolean, String> {
```
Comment (Exclearf) [inline app/src/main/java/com/example/myapplication/CredentialsManager.kt:9]: suggestion(non-blocking): Can we extract these strings into resources?

File: app/src/main/java/com/example/myapplication/RegistrationActivity.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            val isEmailValid = validateField(
+                layout = emailLayout,
+                value = email,
+                errorMessage = "Invalid email format"
+            ) { credentialsManager.isEmailValid(it) }
+
+            val isPasswordValid = validateField(
+                layout = passwordLayout,
+                value = password,
+                errorMessage = "Password must contain at least 8 characters, including uppercase, lowercase, number and special character"
```
Comment (Exclearf) [inline app/src/main/java/com/example/myapplication/RegistrationActivity.kt:51]: suggestion(non-blocking): Yep, same as the comment above. Same string, different place. Looks like a nice candidate for a resource string :D

File: app/src/main/java/com/example/myapplication/RegistrationActivity.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            ) { credentialsManager.isPasswordValid(it) }
+
+            if (isEmailValid && isPasswordValid) {
+                val (success, message) = credentialsManager.register(email, password)
+
+                if (success) {
+                    // Redirect to LoginActivity
+                    val intent = Intent(this, MainActivity::class.java)
```
Comment (Exclearf) [inline app/src/main/java/com/example/myapplication/RegistrationActivity.kt:59]: question(non-blocking): self-documented code documented :P

File: app/src/main/java/com/example/myapplication/MainActivity.kt [hunk 4]
```diff
@@ -52,8 +55,17 @@ class MainActivity : AppCompatActivity() {
...
 
             if (isEmailValid && isPasswordValid) {
+                val intent = Intent(this, EmptyActivity::class.java)
+                startActivity(intent)
+                finish()
             }
         }
+
+        // Navigate to RegistrationActivity when "Register now" is clicked
+        registerText.setOnClickListener {
```
Comment (Exclearf) [inline app/src/main/java/com/example/myapplication/MainActivity.kt:65]: question(non-blocking): Isn't this self-explanatory code?

File: app/src/main/res/layout/activity_empty.xml [hunk 5]
```diff
@@ -6,4 +6,40 @@
...
+        android:layout_height="wrap_content"
+        android:text="You are logged in!"
+        android:textSize="20sp"
+        android:textStyle="bold"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/successIcon"
+        app:layout_constraintVertical_bias="0.1" />
+
+    <!-- Continue Button -->
```
Comment (Exclearf) [inline app/src/main/res/layout/activity_empty.xml:34]: I'd debate whether this is needed as well, but leave it be ;D

**Review by Exclearf — APPROVED** [review]

**Review by Exclearf — APPROVED** [review]

**Comment by Exclearf** [comment]
image

File: app/src/main/java/com/example/myapplication/RegistrationActivity.kt [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            }
+        }
+    }
+
+    private fun validateField(
+        layout: TextInputLayout,
+        value: String,
+        errorMessage: String,
+        validationLogic: (String) -> Boolean
+    ): Boolean {
```
Comment (mtkar-dev) [inline app/src/main/java/com/example/myapplication/RegistrationActivity.kt:75]: It's great
You have a bit of duplication there though, we can make it shorter
1. 
```
val isValid = validationLogic(value)

layout.Error = if(isValid) "" else errorMessage 
layout.isErrorEnabled = !isValid

return isValid
```

2. 
You can check scope functions if you want to make it even cleaner
https://kotlinlang.org/docs/scope-functions.html#function-selection

You could then do something like...


```
return validationLogic(value)
  .apply{isValid ->
		layout.Error = if(isValid) "" else errorMessage 
		layout.isErrorEnabled = !isValid
}
```

or even better yet use a oneliner

**Pull request merged**